olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -96,8 +97,33 @@ class DATASERVICE_READ_API VersionedLayerClient final {
    * @param layer_id The layer ID of the versioned layer from which you want to
    * get data.
    * @param settings The `OlpClientSettings` instance.
+   *
+   * @deprecated Will be removed by 06.2020.
+   */
+  OLP_SDK_DEPRECATED(
+      "Use the ctor with the explicitly specified version. This ctor is "
+      "deprecated and will be removed "
+      "by 06.2020")
+  VersionedLayerClient(client::HRN catalog, std::string layer_id,
+                       client::OlpClientSettings settings);
+  /**
+   * @brief Creates the `VersionedLayerClient` instance with the specified
+   * catalog version.
+   *
```

Comment:
Please add a detailed description stating that:

- If you provide boost::none we will internally query for the latest catalog version and lock the instance to that version.
- That this way we assure data consistency.
- That starting with this we ignore all request structure versions provided by the user.

Reply:
Done - please, double check again